app/scripts/lib/metrics.js

```diff
@@ -303,6 +304,18 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Log an event only if it never happened before during this page load.
+     *
+     * @param {String} eventName
+     */
+    logEventOnce: function (eventName) {
```

Comment:
One case to consider here is:

```js

this.logEvent('eventName');
this.logEventOnce('eventName');
```

When `logEventOnce` is called, should it log an additional time? That seems a bit unexpected. 

If you go with storing a list of events locally, I'd add the eventName to the list in `logEvent` and check whether it's already stored here.